tests/end2end/data/fileselect/single_fileselect.html

```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <title>Testing fileselection</title>
+        <script type="text/javascript">
+            function onLoad() {
+                const input = document.querySelector('#file_upload');
+                input.addEventListener('input', function (e) {
+                    if (input.files.length == 1) {
```

Comment:
Perhaps log something here if the length isn't 1, so that there's some information if that happens rather than a (rather silent) failure? Perhaps that'll also tell you more about why your test isn't working at the moment.

Reply:
Good point, I'll add that!

Reply:
Done